### ISSUE TITLE
Package as rpm/deb and add it to the offical repos?

### ISSUE THREAD

**Issue by mkg20001** [ipfs/go-ipfs#5471]
#### Description: Package go-ipfs as rpm/deb package

This would allow IPFS to be installed with a simple `sudo apt-get install go-ipfs -y` without having to worry about updating, etc.

**Comment by keks** [comment]
Hi @mkg20001,

if you want to install ipfs via your package manager, you will have to file this issue with your distribution. For example, Debian has a [list of prospective packages] that can be picked up for packaging by Debian maintainers. Apparently ipfs has been on that list for over three years now. Anyway, there is not much we can do about this, as this is a job for your distribution's maintainers.

However, there is a snap for ipfs (`snap find ipfs`), but unfortunately [it had some issues] that are currently being resolved, which is why it's still at 0.4.14. But I guess Debian wouldn't distribute more recent versions either ;)

[list of prospective packages]: https://www.debian.org/devel/wnpp/requested.en.html
[it had some issues]: https://github.com/ipfs/go-ipfs/issues/5423

**Comment by mkg20001** [comment]
@keks Ok, I'll leave this issue open until those packages are ready.

**Comment by alexanderkjeldaas** [comment]
Make sure it's safe to run #5511

**Comment by lgierth** [comment]
Let's go with a PPA before attempting to add IPFS to the official Debian/Ubuntu repos. IPFS is far from stable software, and having it in the official repos is only going to lead to a ton of old IPFS nodes out there.

- Debian/Ubuntu packaging :+1: 
- PPA :+1: 
- Official repos :-1:

**Comment by mitra42** [comment]
Yes - we are needing a way to do an automated install for IPFS in order to put it in docker so we can move IPFS into our production environment (at Internet Archive), and were hoping for apt-get as the docker instructions don't work for our situation for a couple of reasons.

**Comment by Kubuxu** [comment]
As we publish full source tarballs packaging it should be easier (see https://dist.ipfs.io/go-ipfs/v0.4.17).

**Comment by mkg20001** [comment]
Something quick and dirty to get debs would be:
```
cd $(mktemp -d)
wget https://dist.ipfs.io/go-ipfs/v0.4.17/go-ipfs_v0.4.17_linux-amd64.tar.gz -O ipfs.tar.gz
tar xvfz ipfs.tar.gz
install -D go-ipfs/ipfs pkg/bin/ipfs
install -D go-ipfs/LICENSE pkg/LICENSE
install -D go-ipfs/README.md pkg/README
cd pkg
tar cvfz ../go-ipfs_v0.4.17_amd64.tar.gz .
cd ..
sudo alien ./go-ipfs_v0.4.17_amd64.tar.gz
```

For the future maybe using a proper `debian/` config directory would be better, although alien can be also used to generate rpms, etc

**Comment by mkg20001** [comment]
Looks like there already is a PPA serving go-ipfs packages: http://get.siderus.io/
// cc @koalalorenzo

**Comment by mcepl** [comment]
One failed attempt to package ``go-ipfs`` for openSUSE is recorded at https://github.com/ipfs/go-ipfs/issues/7881.

**Comment by danger89** [comment]
This shouldn't be that hard I was hoping. Just package the `ipfs` binary for amd64 platform to get started, package it into a deb file (DEBIAN directory with control, postinst files). Add push is via CI/CD to PPA or official package managers.

**Comment by LynHyper** [comment]
@Stebalien Any updates on this issue? A standalone Deb package would be a huge help for new users

**Comment by Stebalien** [comment]
A Debian maintainer would need to pick this up. It should be easier this time around now that go has go modules, but I'm not going to spend any time trying to appease Debian.

My concerns are:

1. Debian deciding to replace dependencies with their own versions instead of the ones we specify (this is what caused the last attempt to fail).
2. Debian deciding to patch go-ipfs because they know better (I really don't want to deal with those bug reports).
3. Debian deciding to not update because stability means "old software" (this is why we haven't bothered pursuing this as I again don't want to deal with these bug reports).

A better approach might be something like rustup: an ipfs-update that gets installed as `/usr/bin/ipfs` and auto-installs the correct go-ipfs version.

**Comment by LynHyper** [comment]
What about having pre-built `.deb` packages as release packages as mentioned by https://github.com/ipfs/ipfs-desktop/issues/691#issuecomment-499302207?

**Comment by Stebalien** [comment]
Hm. Yeah, actually, that seems reasonable. Especially given that we have pre-built other releases.

**Comment by Winterhuman** [comment]
@Stebalien Now that https://github.com/ipfs/ipfs-desktop/#install lists `.deb` and `.rpm` packages, though they're listed as "experimental", should this issue be closed or does the experimental status need to taken care of first?

**Comment by Stebalien** [comment]
Well, that's a release of ipfs-desktop. I assume this is a request for packaging go-ipfs directly (e.g., for servers).
